### ISSUE TITLE
AMD gpu won't boot on UEFI or without nomodeset

### ISSUE THREAD

**Issue by gitbobbed** [linuxdeepin/developer-center#1699]
## Describe the bug

Deepin only reaches a graphical interface with nomodeset and on CSM mode. Issue happens with USB live disk and persists after installation.

## To Reproduce

1st path.
- Boot USB in UEFI mode.
- Choose "Install Deepin" option with options `quiet` and `splash` removed.
- System completely freezes right after `Starting Light Display Manager`. Sometimes one or more tasks will complete after that and before freeze, but I never see a graphical interface. No error message is shown. USB stick gets completely removed from system and won't be detected again until removed from the port and reinserted (even if I turn the whole PC off and on again).

2nd path.
- Boot USB in EUFI mode.
- Chose "Install Deepin" option with options `quiet` and `splash` removed and with `nomodeset` added.
- System shows a segfault error right before a graphical interface would appear (see screenshot). 
No alternative terminal can be reached with Alt + F1 etc., but can still press ctrl + alt + delete and reboot the system.

3rd path.
- Boot USB in CSM (legacy) mode.
- Chose "Install Deepin" option with options `quiet` and `splash` removed and with `nomodeset` added.
- System boots normally. Can install Deepin, but it does not install ESP to the EFI partition (even though it is explicitly told to do it during manual install). Only boots GRUB on legacy mode.
- System gets sluggish after install. Interface stutters and frame rate is quite low (result of nomodeset).
- Have tried updating the kernel to 5.3.18 and 5.4.28 and manually downloading all firmware files from kernel source for amdgpu and copying them to /lib/firmware. Still won't work without nomodeset. 

 ## Expected behavior

## Screenshots

Boot result in UEFI with `nomodeset` added to boot arguments
![image](https://user-images.githubusercontent.com/33555741/80165183-344cc580-85db-11ea-8e83-3e3f5b1227f7.png)

### Distro & Version

Linux Deepin v20 Beta
All updates installed
### Additional context

* Have tried getting different ISOs and MD5 checks out.
* Have tried manual and full-hard-disk install. Same issue.
* Have tried installing to a different hard disk. Same issue.
* Have tried using multiple apps to burn the ISO to the pendrive (etcher, Rufus, WoeUSB, deepin-boot-maker and Yumi). None of them reaches a graphical interface except for using the 3rd path above.
* Same ISO installs fine on Virtualbox
* Several other Linux distributions install just fine on the same system and hard disk(s).

Here are my system specs:

```
|          [CPU][AMD Ryzen 5 3600 3.6 GHz 6-Core Processor]
|   [CPU Cooler][Corsair H60 (2018) 57.2 CFM Liquid CPU Cooler]
|  [Motherboard][Gigabyte X570 AORUS PRO ATX AM4 Motherboard]
|       [Memory][Corsair Vengeance LPX 32 GB (2 x 16 GB) DDR4-3200 Memory]
|      [Storage][Samsung 850 EVO-Series 500 GB 2.5" Solid State Drive]
|      [Storage][Samsung 860 Evo 500 GB 2.5" Solid State Drive]
|      [Storage][Samsung 970 Evo 500 GB M.2-2280 NVME Solid State Drive]
|   [Video Card][PowerColor Radeon RX 5700 XT 8 GB Red Devil Video Card]
|         [Case][Cooler Master MasterCase H500P Mesh ATX Mid Tower Case]
| [Power Supply][Corsair RMx (2018) 750 W 80+ Gold Certified Fully Modular ATX Power Supply]
|      [Monitor][MSI Optix MAG341CQ 34.0" 3440x1440 100 Hz Monitor]
|       [Custom][ABWB 802,11 AC WI-FI + Bluetooth 4.0 PCI-Express (PCI-E) BCM94360CD Combo Karte für Hackintosh (Mac OS X) Flughafen Computer]
|       [Custom][Creative Sound Blaster Omni Surround 5.1 USB Sound Card with High Performance Headphone Amp and Integrated Microphone]
```

**Comment by DanyGee** [comment]
Can confirm exactly the same problems on GTX1080
![20200415_171953](https://user-images.githubusercontent.com/26818551/80289127-fbc4fd00-873c-11ea-982b-0eba57a0506e.jpg)

My rig:
![20200415_171551](https://user-images.githubusercontent.com/26818551/80289132-097a8280-873d-11ea-9999-c8eb0033cad0.jpg)

**Comment by gokhanyildirimlar** [comment]
Can you take a look at this, @xmuli? This seems like affecting quite a few people and they can't even install Deepin 20 Beta on their hardware. This issue requires a certain fix before the stable release arrives because installation image will not be altered for some time after release.

**Comment by jouyouyun** [comment]
@gitbobbed please provide the `/var/log/lightdm` directory and the `/var/log/Xorg.0.log` file in a problem environment, thanks!

**Comment by DanyGee** [comment]
While the original creator sleeps :wink: , I'll gladly provide logs from my environment (GTX1080).
While booting I deleted the `quiet`, `splash` and `installer` options and added `nomodeset` to be able to actually do something.
Please find the logs attached below:

:card_file_box: 
[LOGS_GTX1080_DEEPINv20.zip](https://github.com/linuxdeepin/developer-center/files/4896001/LOGS_GTX1080_DEEPINv20.zip)

**Comment by gokhanyildirimlar** [comment]
Uh! So installer doesn't work but live-cd works? Strange behavior. @jouyouyun

> While the original creator sleeps  , I'll gladly provide logs from my environment (GTX1080).
> While booting I deleted the `quiet`, `splash` and `installer` options and added `nomodeset` to be able to actually do something.
> Please find the logs attached below:
> 
> 
> [LOGS_GTX1080_DEEPINv20.zip](https://github.com/linuxdeepin/developer-center/files/4896001/LOGS_GTX1080_DEEPINv20.zip)

Uh! So installer doesn't work but live-cd works? Strange behavior. @jouyouyun

**Comment by DanyGee** [comment]
> Uh! So installer doesn't work but live-cd works? Strange behavior. @jouyouyun
> 
> > While the original creator sleeps  , I'll gladly provide logs from my environment (GTX1080).
> > While booting I deleted the `quiet`, `splash` and `installer` options and added `nomodeset` to be able to actually do something.
> > Please find the logs attached below:
> > [LOGS_GTX1080_DEEPINv20.zip](https://github.com/linuxdeepin/developer-center/files/4896001/LOGS_GTX1080_DEEPINv20.zip)
> 
> Uh! So installer doesn't work but live-cd works? Strange behavior. @jouyouyun

No, both v20 installer and live cd don't work.
On default boot the video signal gets lost.
Removing installer and adding `nomodeset` results in hang during boot, just like on screenshots above.
But `nomodeset` lets me switch to tty2 with <kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>F2</kbd>.
And here, in text mode I was able to copy the logs.
Both Live CD and installer from 15.11 works without problems with nomodeset - no boot hang.

**Comment by jouyouyun** [comment]
```shell
 [    15.156] (EE) [drm] Failed to open DRM device for pci:0000:01:00.0: -19 
 [    15.156] (EE) open /dev/dri/card0: No such file or directory
 [    15.156] (WW) Falling back to old probe method for modesetting
 [    15.156] (EE) open /dev/dri/card0: No such file or directory
 [    15.156] (II) Loading sub module "fbdevhw"
 [    15.156] (II) LoadModule: "fbdevhw"
 [    15.156] (II) Loading /usr/lib/xorg/modules/libfbdevhw.so
 [    15.156] (II) Module fbdevhw: vendor="X.Org Foundation"
 [    15.156]    compiled for 1.20.4, module version = 0.0.2
 [    15.156]    ABI class: X.Org Video Driver, version 24.0
 [    15.156] (EE) Unable to find a valid framebuffer device
 [    15.156] (WW) Falling back to old probe method for fbdev
 [    15.156] (II) Loading sub module "fbdevhw"
 [    15.156] (II) LoadModule: "fbdevhw"
 [    15.156] (II) Loading /usr/lib/xorg/modules/libfbdevhw.so
 [    15.156] (II) Module fbdevhw: vendor="X.Org Foundation"
 [    15.156]    compiled for 1.20.4, module version = 0.0.2
 [    15.156]    ABI class: X.Org Video Driver, version 24.0
 [    15.157] (II) FBDEV(2): using default device
 [    15.157] (EE) Screen 0 deleted because of no matching config section.
 [    15.157] (II) UnloadModule: "modesetting"
 [    15.157] (EE) Screen 0 deleted because of no matching config section.
 [    15.157] (II) UnloadModule: "fbdev"
 [    15.157] (II) UnloadSubModule: "fbdevhw"
 [    15.157] (EE)
 Fatal server error:
 [    15.157] (EE) Cannot run in framebuffer mode. Please specify busIDs        for all framebuffer devices
 ```

Maybe the graphic card unrecognized, we will build the environment internally for testing.

**Comment by chaim1221** [comment]
As mentioned in #1841 I am unable to boot on a Lenovo Yoga 720 with an _Nvidia_ GPU. The text is very small so it's hard to edit the grub options but I tried the workaround here (with and without `nomodeset`) and it didn't work. I'll probably open a new issue, but I was wondering if it is related. Will link this issue.

**Comment by chaim1221** [comment]
That wasn't the issue. Same result. I think I reloaded the defaults at some point.

**Comment by jouyouyun** [comment]
Try this：https://www.deepin.org/zh/

**Comment by Schykle** [comment]
It's now December 29th, and I have the same issues described here with my RX 5700 XT. That's an AMD card BTW, which both kernels 5.4 and 5.8 should support. Yet I can't even boot into the installer, it just gets stuck at the deepin logotype and nothing ever happens.

**Comment by rdiezsj** [comment]
same problem here, with Ryzen 2600 and RX5600XT. This is the only distro that don´t boot in my hardware.

**Comment by Amachik** [comment]
Same problem with amd 5 3600 and rx 5600xt

**Comment by breizh2944** [comment]
Hi, 
Same here, with fresh 20.2 release (5.10 or 5.11 kernel - but did the same with 5.7 and 5.8 also).
Cannot boot on Live CD or "real" install if I don't choose safe graphics (it only works with nomodeset).
CPU AMD Ryzen 9 and RX 5700 XT.

I guess it can affect lots of people... so this is a real big issue, that seems to happen for a lot of time now.
Hoping you will take it into account quite quickly now.

**Comment by rdiezsj** [comment]
This problem should be corrected as soon as possible, since these AMD GPUs should be supported directly by the kernel

**Comment by gitbobbed** [comment]
Sorry for not coming back to this soon enough, it took so many months to get a response that I didn't see the request for further data. Thank you @DanyGee for taking up reporting duty :).

I have since stopped using Deepin (gave up on it), so cannot test further, unfortunately.

FTR, this may have to do with the fact that Debian doesn't ship the necessary firmware files for NAVI GPUs by default (or at least didn't when I last tested it six months ago). 

To run Debian with amdgpu I had to download the necessary firmware files from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/amdgpu.

Not really sure if that's really related, but at least something to investigate.

**Comment by breizh2944** [comment]
Hi,
I just tried the 20.2.1 release, and both install live cd and hard drive install now work.
Hope this is fixed for you also.

**Comment by Zeno-sole** [comment]
We have updated the kernel and firmware on 20.3, did this solve your problem? @gitbobbed

**Comment by gitbobbed** [comment]
> We have updated the kernel and firmware on 20.3, did this solve your problem? @gitbobbed

Hey there @Zeno-sole,

Now that's a blast from the past! Thank you for getting back to it.

I haven't used deepin since little after I posted this issue, actually, but I just tried again with the 20.3 iso and it seems to have worked. 

It booted straight to the GUI without any modifications and installed the system fine (even though it took a long time to do it, like, unusually long compared to any of the other several distros I have installed in the same machine).

I say 'seems' to have worked only because I couldn't get past the first boot settings screen (it says that it is applying my settings but just stayed there for a good 10-15 minutes and nothing happened), but since both the live ISO and the installed system booted all the way to a graphical interface without having to tweak any boot parameters in GRUB, I'm guessing the driver issue has been resolved (at least for 5000 series GPUs).

Best,

Daniel

**Comment by Zeno-sole** [comment]
Are you saying that the installation time is too long?

**Comment by gitbobbed** [comment]
> Are you saying that the installation time is too long?

Yeah. Took really long. Can't gauge what would be normal other than other distros.

But then it did install, but I couldn't get past the first boot welcome setup.

Got stuck in the last step for hours.

**Comment by Zeno-sole** [comment]
Will it work on non-i915 devices?

**Comment by Zeno-sole** [comment]
Did you get stuck for a few hours after setting up the user?

**Comment by Zeno-sole** [comment]
Due to no reply for a long time and lack of necessary information, close the issue, If the problem still exists, you can apply to reopen the issue

**Comment by st1vms** [comment]
I'm posting my experience so far when installing Ubuntu Jammy LTS with an rx7900xt.
So the experience out of the box is absolutely terrible...first thing first I had problems when installing Ubuntu without safe graphics, as the cursor became invisible in live env. But when installing through safe graphics (mouse cursor showing), I had a similar boot hang problem.

Going through the installation in normal mode (with cursor invisible), I was able to boot successfully into my system.
After installing the very first suggested updates from Software&Updates popup (with xserver amdgpu listed as upgrade), the same bug appeared on reboot.

In order to fix this I had to boot with `nomodeset` kernel parameter, and install GPU drivers using the `amdgpu-install` utility, after installing `dkms` package, required by dkms amdgpu packages installation.

Probably if installed before running suggested updates, bug may not appear right away.
